BungeeCord-Patches/0064-Adventure.patch

```diff
@@ -0,0 +1,1668 @@
++            <artifactId>adventure-text-serializer-bungeecord</artifactId>
++            <version>4.0.0-SNAPSHOT</version>
++        </dependency>
++        <!-- Waterfall end -->
+     </dependencies>
+ </project>
+diff --git a/api/src/main/java/net/md_5/bungee/api/CommandSender.java b/api/src/main/java/net/md_5/bungee/api/CommandSender.java
+index d5888311..72f3da4c 100644
+--- a/api/src/main/java/net/md_5/bungee/api/CommandSender.java
++++ b/api/src/main/java/net/md_5/bungee/api/CommandSender.java
+@@ -3,7 +3,7 @@ package net.md_5.bungee.api;
+ import java.util.Collection;
+ import net.md_5.bungee.api.chat.BaseComponent;
+ 
+-public interface CommandSender
++public interface CommandSender extends net.kyori.adventure.audience.Audience // Waterfall - we're going on an adventure!
+ {
+ 
+     /**
+@@ -34,14 +34,18 @@ public interface CommandSender
+      * Send a message to this sender.
+      *
+      * @param message the message to send
++     * @deprecated in favour of {@link net.kyori.adventure.audience.Audience#sendMessage(net.kyori.adventure.text.Component)}
+      */
++    @Deprecated // Waterfall - we're going on an adventure!
+     public void sendMessage(BaseComponent... message);
+ 
+     /**
+      * Send a message to this sender.
+      *
+      * @param message the message to send
++     * @deprecated in favour of {@link net.kyori.adventure.audience.Audience#sendMessage(net.kyori.adventure.text.Component)}
+      */
++    @Deprecated // Waterfall - we're going on an adventure!
+     public void sendMessage(BaseComponent message);
+ 
+     /**
+diff --git a/api/src/main/java/net/md_5/bungee/api/ProxyServer.java b/api/src/main/java/net/md_5/bungee/api/ProxyServer.java
+index a4011335..092aaf09 100644
+--- a/api/src/main/java/net/md_5/bungee/api/ProxyServer.java
++++ b/api/src/main/java/net/md_5/bungee/api/ProxyServer.java
+@@ -17,7 +17,7 @@ import net.md_5.bungee.api.plugin.Plugin;
+ import net.md_5.bungee.api.plugin.PluginManager;
+ import net.md_5.bungee.api.scheduler.TaskScheduler;
+ 
+-public abstract class ProxyServer
++public abstract class ProxyServer implements net.kyori.adventure.audience.ForwardingAudience // Waterfall - We're going on an adventure!
+ {
+ 
+     @Getter
+@@ -170,9 +170,20 @@ public abstract class ProxyServer
+      * Gracefully mark this instance for shutdown.
+      *
+      * @param reason the reason for stopping. This will be shown to players.
++     * @deprecated in favour of {@link #stop(net.kyori.adventure.text.Component)}
+      */
++    @Deprecated // Waterfall - we're going on an adventure!
+     public abstract void stop(String reason);
+ 
++    // Waterfall start - we're going on an adventure!
++    /**
++     * Gracefully mark this instance for shutdown.
++     *
++     * @param reason the reason for stopping. This will be shown to players.
++     */
++    public abstract void stop(final net.kyori.adventure.text.@org.checkerframework.checker.nullness.qual.NonNull Component reason);
```

Comment:
Imports are fine, for the record. Don't need to change it, just funny to see this.

Reply:
Yeah blame @kashike for that :slightly_smiling_face: